### DISCUSSION TITLE
Supporting the zero point release a bit after releasing 1.0

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#4772] [Q&A]
@ganigeorgiev I'm thinking that it would be very helpful for the adoption of the 1.0 if you can continue updating critical things about the 0.x release for a bit after releasing 1.0.  The reason is that it sounds like that it will be a very disruptive upgrade.  This will allow people to upgrade as late as possible if they cannot afford the risk.  I'd use my hobby project for at least several weeks to flesh out any issues.  There is zero risk there, and I'm also using the JS side as well.

It would be nice if you could backport any critical fixes or Golang or package updates for a bit so that the more adventurous people can pave the way.  Already the 0.x series is effectively a 1.0, and many people are seriously using it.

Another strategy could be to release the massively refactored models and forms code with some minimal low-risk automatic migrate script, and to let people settle into that for some weeks before you release some large PB Admin migrations that could hurt people's projects.  It might benefit from having the changes *mostly* separated into a few stages.  So in this model you would release more 0.x major versions.

**Comment by ganigeorgiev** [comment]
I think there is some misunderstanding. There will be at least several more v0.x releases before a stable v1.0.

The [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) is planned to be released as v0.23.0 (_at least the backend part_).

It will mainly introduce Go (and JS `pb_hooks`) API changes and there is just no easy way around this (_hopefully it will be the last major refactoring_). The refactoring cannot be released on several stages because everything is tightly coupled and/or depend on some common part to work. I'm still experimenting so there is nothing finalized yet that I could show but one example is that everything will be "hook", including the middlewares and routes, aka. they will share the same declaration `func (e *SomeEvent) error`.

Similar to the v0.8 refactoring, there will be a helper command that will migrate existing data to the new structure and a detailed guide with the changes but userland Go and JS `pb_hooks` code will need to be upgraded manually.

There will not be major breaking changes to the Web APIs. The batch support works with the existing actions by emulating an internal request (_this is done mainly so that one request hook will work the same for both regular and batch requests_). The main Web API breaking changes so far are the collections schema, admins (converted to `superusers` system auth collection) and the `/auth-methods` endpoint due to the MFA support and the changes related to the password identity fields and the OAuth2 improvements (_I'll see if I can squeeze OTP/TOTP too as there were several requests for it but I'll think more on it after most of the other tasks are finalized_).

I'll consider maintaining the current v0.22.x in its own separate branch for some time after the v0.23.x release, but it will not receive new features, only security and blocker bug fixes.

**Reply by dschissler** [reply]
So it seems that it won't be much different in nature to the 0.8 update that introduced all collections being able to use authentication, except that you'll support the previous paradigm for a bit.
